### PR TITLE
Flush ansi color context after printing

### PR DESCRIPTION
**Pull request by dpsutton**
Before submitting the PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines][1]
- [x] You've added tests (if possible) to cover your change(s)
- [x] All tests are passing (`make test`)
- [x] The new code is not generating bytecode or `M-x checkdoc` warnings
- [ ] You've updated the changelog (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)
- [ ] You've updated the refcard (if you made changes to the commands listed there)

References https://github.com/clojure-emacs/cider/issues/1794

### CODE REVIEW

**Comment by Malabarba** [comment]
Answering the question on the issue: I like the way it's implemented here (a function rather than inline code), and I think it shouldn't cause much problem (at least, not more problem than the current code).

However, it looks like you broke: ./test/cider-util-tests.el:141:

File: cider-repl.el [hunk 1]
```diff
@@ -526,6 +526,14 @@ Return the position of the prompt beginning."
         (set-marker cider-repl-prompt-start-mark prompt-start)
         prompt-start))))
 
+(defun cider-repl--flush-ansi-color-context ()
+  "Flush ansi color context after printing.  When there is a possible
```
Comment (bbatsov) [inline cider-repl.el]: Add a line break after the first sentence. Standard Emacs documentation convention.

File: test/cider-repl-tests.el [hunk 2]
```diff
@@ -68,3 +68,20 @@
 ;;  Javadoc: (javadoc java-object-or-class)
 ;;     Exit: <C-c C-q>
 ;;  Results: Stored in vars *1, *2, *3, an exception in *e;"))))
+
+(defun create-cider-string (s)
```
Comment (bbatsov) [inline test/cider-repl-tests.el]: This could use some docstring as it's not immediately obvious what it does.

Comment (bbatsov) [inline test/cider-repl-tests.el]: Maybe this should be named `cider-create-repl-output-string`?

**Comment by dpsutton** [comment]
Ok, fixed the logic but need to move my test. I've got to head out for a bit though and I'll do it when i get back.

thanks for the heads up.

**Pull request merged**